arch/cortex-m/src/support.rs

```diff
@@ -47,6 +47,26 @@ where
     return res;
 }
 
+#[cfg(target_os = "none")]
+pub unsafe fn atomic_write(location: &mut u64, value: u64) {
+    // Set PRIMASK
```

Comment:
There is already an atomic primitive in this module, so both of these could (should?) simply be re-written as:

```rust
atomic(|| { *location = value });
```


```rust
atomic(|| { *location });
```

To other notes:

  * In general exposing atomic operations by turning off interrupts is dangerous because they _cannot_ be nested or interleaved in any way. E.g., if you already happen to be in an "atomic" block, and perform an `atomic_read`, it will turn interrupts back on prematurely. This is only to say that we actually ought to be very careful about using these primitives, and how they are exposed.

  * The loads and stores of `location` should probably be volatile to make sure the compiler doesn't elide them if it thinks the change to `location` is not used, or move them if the result isn't actually used until later. So, maybe `::core::ptr::write_volatile(location, value)` and `::core::ptr::read_volatile(location)`.

Reply:
Thanks. Cleaned this up